internal/engine/experiment/torsf/torsf.go

```diff
@@ -73,22 +94,27 @@ func (m *Measurer) Run(
 	ctx context.Context, sess model.ExperimentSession,
 	measurement *model.Measurement, callbacks model.ExperimentCallbacks,
 ) error {
+	ptl, sfdialer, err := m.setup(sess.Logger())
+	if err != nil {
+		// we cannot setup the experiment
+		return err
+	}
+	defer ptl.Stop()
```

Comment:
We recently said that sync tests should only return error when something fundamental happens and the test could not run. For this reason, I've refactored `torsf` so that first we see if all the preconditions are met. If they are not met, then we just return an error. Otherwise, we measure. This design is cleaner with respect to the changing requirements in terms of erroring out or returning nil.